middleware.js

```diff
@@ -1,34 +1,108 @@
 const vueCompiler = require('@vue/component-compiler')
+const parseUrl = require('parseurl')
 const { transformModuleImports } = require('./transformModuleImports')
-const { loadPkg } = require("./loadPkg")
+const { loadPkg } = require('./loadPkg')
 const { readSource } = require('./readSource')
 
-const vueMiddleware = root => {
+const defaultOptions = {
+  cache: true
+}
+
+const vueMiddleware = (options = defaultOptions) => {
+  let cache
+  if (options.cache) {
+    const LRU = require('lru-cache')
+
+    cache = new LRU({
+      max: 100,
+      length: function (n, key) { return n * 2 + key.length },
+      maxAge: 1000 * 60 * 60
+    })
+  }
+
   const compiler = vueCompiler.createDefaultCompiler()
 
   function send(res, source, mime) {
     res.setHeader('Content-Type', mime)
     res.end(source)
   }
 
+  function tryCache (key) {
+    return cache.get(key)
+  }
+
+  function cacheData (key, data) {
+    const old = cache.peek(key)
+
+    if (old != data) {
+      cache.set(key, data)
+      return true
+    } else return false
+  }
+
+  async function bundleSFC (req) {
+    const { filepath, source } = await readSource(req)
+    const descriptorResult = compiler.compileToDescriptor(filepath, source)
+    return vueCompiler.assemble(compiler, filepath, descriptorResult)
+  }
+
   return async (req, res, next) => {
-    // TODO caching
     if (req.path.endsWith('.vue')) {
-      const { filepath, source } = await readSource(req)
-      const descriptorResult = compiler.compileToDescriptor(filepath, source)
-      const assembledResult = vueCompiler.assemble(compiler, filepath, descriptorResult)
+      const key = parseUrl(req).pathname
```

Comment:
The cache for a file needs to be busted whenever the source code changes (user hits save, file written to disk) - simple implementation would be using `fs.stats` to get the last modified timestamp and store that along with the compile result in the cache.